pytorch_lightning/overrides/torch_distributed.py

```diff
@@ -88,7 +88,7 @@ def _broadcast_object_list(object_list, src=0, group=None):
             object_list[i] = _tensor_to_object(obj_view, obj_size)
 
 
-if _TORCH_GREATER_EQUAL_1_7 and torch.distributed.is_available():
+if _TORCH_GREATER_EQUAL_1_8 and torch.distributed.is_available():
```

Comment:
I don't understand the fix. `_broadcast_object_list` is taken from 1.7.1

https://github.com/PyTorchLightning/pytorch-lightning/blob/20f63377f81f4771d3f128f979b3a0f9b8d219a7/pytorch_lightning/overrides/torch_distributed.py#L43-L44

You say in the linked issue:

> but only for versions older than 1.7. Unfortunately, the problem persists in all of 1.7.x.

`_TORCH_GREATER_EQUAL_1_7` should be true in 1.7.x so we should be using the same implementation as the code was copy-pasted.

Reply:
> I don't understand the fix. _broadcast_object_list is taken from 1.7.1

I think that comment is misleading... It should probably read "Taken from ... and modified"

This is how the original code handles the `to` call:
https://github.com/pytorch/pytorch/blob/1.7/torch/distributed/distributed_c10d.py#L1368-L1369

But this is what the override function does:
https://github.com/PyTorchLightning/pytorch-lightning/blob/20f63377f81f4771d3f128f979b3a0f9b8d219a7/pytorch_lightning/overrides/torch_distributed.py#L59-L65

> _TORCH_GREATER_EQUAL_1_7 should be true in 1.7.x 
Right. This is the problem. The override should not happen if Torch is 1.7.x. If Torch is above 1.7.x (i.e. 1.8.0 and up), you can use the Torch function.